src/http/web_socket/protocol.cr

```diff
@@ -326,7 +328,7 @@ class HTTP::WebSocket::Protocol
 
   def self.key_challenge(key)
     {% if flag?(:without_openssl) %}
-      Digest::SHA1.base64digest(key + GUID)
+      ::Crystal::Digest::SHA1.base64digest(key + GUID)
```

Comment:
IIUC this won't work without `: without_openssl` flag.

Reply:
Exactly. `-Dwithout_openssl` will use the 100% crystal sha1 implementation. But by default the openssl one is used. The only thing that is changing here is the location of the crystal sha1 implementation.

Reply:
How come? Does `::Crystal::Digest::SHA1` delegates to `OpenSSL::...` somehow?

Reply:
`::Crystal::Digest::SHA1` is the pure crystal implementation (formerly `::Digest::SHA1`).

After this PR `::Digest::SHA1` uses openssl, but `::Crystal::Digest::SHA1` does not.

Reply:
That's what I'm asking about since the `require` is conditional.

Reply:
So is the instantiation.

Reply:
@straight-shoota So true, somehow I missed that 🤦‍♂️